### PR TITLE
snapd: purchasing REST API proposal

### PR DESCRIPTION
**Pull request by alecu**
This is a proposal of how the REST api would look like after adding support for making purchases

### CODE REVIEW

**Comment by snappy-m-o** [comment]
Can one of the admins verify this patch?

File: docs/rest.md [hunk 1]
```diff
@@ -329,7 +336,7 @@ See `sources` for `/2.0/snaps`.
 
 field      | ignored except in action | description
 -----------|-------------------|------------
-`action`   |                   | Required; a string, one of `install`, `update`, `remove`, `purge`, `activate`, `deactivate`, or `rollback`.
```
Comment (niemeyer) [inline docs/rest.md]: s/purchase/buy/

File: docs/rest.md [hunk 2]
```diff
@@ -329,7 +336,7 @@ See `sources` for `/2.0/snaps`.
...
+`action`   |                   | Required; a string, one of `purchase`, `install`, `update`, `remove`, `purge`, `activate`, `deactivate`, or `rollback`.
```
Comment (niemeyer) [inline docs/rest.md]: s/purchase/buy/

File: docs/rest.md [hunk 3]
```diff
@@ -222,9 +226,10 @@ Sample result:
 
 #### Fields
 * `snaps`
-    * `status`: can be either `not installed`, `installed`, `active` (i.e. is
-      current), `removed` (but data present); there is no `purged` state, as a
-      purged snap is undistinguishable from a non-installed snap.
+    * `status`: can be either `buyable`, `available`, `installed`,
```
Comment (niemeyer) [inline docs/rest.md]: Buyable is such a terrible term.. I couldn't come up with something better in our conversation, but now I actually have another suggestion for these sequence of events. Here is the full suggested text:

    * `status`: may transition as `available` => `installed` => `active`. For paid snaps,
       the initial state is `priced` and once bought it becomes `available`.

File: docs/rest.md [hunk 4]
```diff
@@ -311,8 +318,8 @@ See `sources` for `/2.0/snaps`.
 
 ### POST
 
-* Description: Install, update, remove, purge, activate, deactivate, or
-  rollback the snap
+* Description: Purchase, install, update, remove, purge, activate, deactivate,
```
Comment (niemeyer) [inline docs/rest.md]: s/purchase/buy/, and please move the action to the end. We don't want to emphasize it, looking like every snap needs to be bought.

**Comment by niemeyer** [comment]
@chipaca?

File: docs/rest.md (hunk 3 above)
Comment (chipaca) [inline docs/rest.md]: Why is buyable “such a terrible term”? It's synonymous with “for sale” or “purchasable”, and you're asking that we use “buy” as the verb of the action that moves the snap from this state to available. It seems a natural fit.

File: docs/rest.md [hunk 5]
```diff
@@ -240,6 +245,8 @@ Sample result:
     * `update_available`: if present and not empty, it means the snap can be
       updated to the version specified as a value to this entry.
     * `channel`: which channel the package is currently tracking.
+    * `prices`: if present the snap needs to be purchased before installing. It's
```
Comment (chipaca) [inline docs/rest.md]: if we're going with `buy` as the verb, here we want “bought” instead of “purchased”.

File: docs/rest.md [hunk 6]
```diff
@@ -240,6 +245,8 @@ Sample result:
...
+      a dict with the available currency codes as keys and the prices as values.
```
Comment (chipaca) [inline docs/rest.md]: it's not a dict, because json isn't python :-) it's an object which member names are the ISO 4217 currency codes etc etc.

**Comment by snappy-m-o** [comment]
Can one of the admins verify this patch?

**Comment by snappy-m-o** [comment]
Can one of the admins verify this patch?

**Comment by chipaca** [comment]
Closing because it's old and I believe superseded by pete's work.

**Pull request closed without merging**